### DISCUSSION TITLE
Bug in RandomLinkSplit

### DISCUSSION THREAD

**Discussion by dpaysan** [pyg-team/pytorch_geometric#3267] [Q&A] [answered]
Dear all,

I had some issues with some dimension mismatches and thereby saw that after determining the random split indices in the RandomLinkSplit finally the data is split not as I had expected. In the RandomLinkSplit class the data splits actually do not use the sampled validation and test edges but reuse the train edges for the validation data and a combination of the training and validation edges for the test data. I have copied the respective part of the code below which can be found [here](https://github.com/pyg-team/pytorch_geometric/blob/a2d313dcfe8ed1012061c9aebf329d35e86c4db8/torch_geometric/transforms/random_link_split.py#L130) :

        train_edges = perm[:num_train]
        val_edges = perm[num_train:num_train + num_val]
        test_edges = perm[num_train + num_val:]
        train_val_edges = perm[:num_train + num_val]

        # Create data splits:
        train_data = self._split(data, train_edges)
        val_data = self._split(data, train_edges)
        test_data = self._split(data, train_val_edges)``

Could somebody explain why that is?

Thanks a lot in advance!

**Comment by rusty1s** [comment, marked as answer]
`data.edge_index` refers to the edges that are used for message passing. As such, during training and validation, you are allowed to propagate information based on the training edges, while during testing, you can propagate information based on the union of training and validation edges.

For evaluation, the `data.edge_label_index` holds a batch of positive and negative samples that should be used to evaluate your model on.

So:
* `data.edge_index` should be solely used for message passing
* `data.edge_label_index` should be used for evaluation

**Reply by dpaysan** [reply]
That makes sense. Thanks!
